File: t/PACSFramework/VB/StorageServerManager/UI/AdministrativeSettings/Users/UserView.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.Medical.Winforms

Namespace Leadtools.Demos.StorageServer.UI
   Partial Public Class UserView : Inherits UserControl
      Private _UsersAccounts As UsersAccountView

      Public ReadOnly Property UsersAccounts() As UsersAccountView
         Get
            Return _UsersAccounts
         End Get
      End Property

      Public Sub New()
         InitializeComponent()

         _UsersAccounts = New UsersAccountView()
         _UsersAccounts.Dock = DockStyle.Fill
         Controls.Add(_UsersAccounts)
         AddHandler _UsersAccounts.ValueChanged, AddressOf _UsersAccounts_ValueChanged
      End Sub

      Private Sub _UsersAccounts_ValueChanged(ByVal sender As Object, ByVal e As EventArgs)
         If Not SettingsChangedEvent Is Nothing Then
            RaiseEvent SettingsChanged(sender, e)
         End If
      End Sub

      Public Event SettingsChanged As EventHandler

   End Class
End Namespace

File: t/PACSFramework/CS/StorageServerManager/UI/AdministrativeSettings/Users/UserView.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Leadtools.Medical.Winforms;

namespace Leadtools.Demos.StorageServer.UI
{  
   public partial class UserView : UserControl
   {
      private UsersAccountView _UsersAccounts;

      public UsersAccountView UsersAccounts
      {
         get { return _UsersAccounts; }         
      }

      public UserView()
      {
         InitializeComponent();

         _UsersAccounts = new UsersAccountView();
         _UsersAccounts.Dock = DockStyle.Fill;
         Controls.Add(_UsersAccounts);
         _UsersAccounts.ValueChanged += new EventHandler(_UsersAccounts_ValueChanged);
      }

      void _UsersAccounts_ValueChanged(object sender, EventArgs e)
      {
         if (SettingsChanged != null)
         {
            SettingsChanged(sender, e);
         }
      }

      public event EventHandler SettingsChanged;


   }
}
